insights/config/specs.py

```diff
@@ -148,6 +148,7 @@
                                     PatternSpec(r"usr/lib64/ImageMagick-6.5.4/config/policy\.xml"),  # ImageMagick 6.5 on rhel-6 64bit.
                                     PatternSpec(r"usr/lib/ImageMagick-6.5.4/config/policy\.xml")]),  # ImageMagick 6.5 on rhel-6 32bit.
     "init.ora"                  : SimpleFileSpec("{ORACLE_HOME}/dbs/init.ora"),
+    "initscript"                : PatternSpec(r"etc/rc\.d/init\.d/.*"),
```

Comment:
`PatternSpec` isn't exactly a normal regular expression.  The dirname and basename are split and only the basename may be a regular expression.  This is an unfortunate implementation detail of the current client that we're still dealing with.  You should change this to:

```
PatternSpec(r"etc/rc.d/init.d/.*"),
```

Reply:
I can confirm, with branch 1.x at least that PatternSpec is a regular expression:
- if I use PatternSpec(r"etc/rc.d/init.d/.*"), then etc/rcXd/initXd/something also matchs, which I don't want to
- if I use PatternSpec(r"etc/rc\.d/init\.d/.*"), it perfectly works: only etc/rc.d/init.d/* matches

Reply:
Sorry I managed to mis-explain it.  When the client collects the spec it doesn't treat it as a regular expression.  When we try to extract it from the archive we do treat it as a regular expression.

Reply:
So what should I do, change to r"etc/rc.d/init.d/.*" anyway?

Reply:
Since this spec is designed to work with sosreports explicitly, leaving it as is will work.  We should wrap the spec with a NoneGroup so that we don't attempt to collect it from customers.  There is an open issue #594 to track the differences between sosreport collection and insights archive collection.